samples/v1/types_config.go

```diff
@@ -64,6 +64,30 @@ type ConfigSpec struct {
 	// content but the operator will not recreate(or update) anything
 	// listed here.
 	SkippedTemplates []string `json:"skippedTemplates,omitempty" protobuf:"bytes,6,opt,name=skippedTemplates"`
+
+	// helmChartList specifies the list of the helmcharts(S2i) that
```

Comment:
are users going to be able to edit this list?  What happens if a non-s2i chart is put in the list?

what happens when an item is deleted from this list? is the chart uninstalled from the cluster?

in general how is this list managed?

Reply:
Users will be able to edit the list, if an item is deleted, chart wont be unistalled, but we wont manage it (same as current samples) 
We will have validation in place where we will only allow S2i helmcharts to be installed/managed by Samples and wont install NonS2i's

Reply:
how are helmcharts identified as "s2i" helm charts?

what will happen if they specify a non-s2i helm chart? (will there be some status message/condition that makes it clear to the user that the chart is not going to be installed?)

whose permissions are used to install the chart?  (can a user leverage this api to elevate their own permissions by using the samples operator to instantiate a helmchart the user cannot instantiate themselves?)

Reply:
The imagestreams has a tag as s2i, and we have identified the 12 which are required by the ODC team for day 1 user experience, so that 12 corresponding helmcharts only will allowed.
If they specifiy a Non-S2i, we plan on displaying a message that it won't be installed

Reply:
> we plan on displaying a message that it won't be installed

displaying a message where?

Reply:
> whose permissions are used to install the chart? (can a user leverage this api to elevate their own permissions by using the samples operator to instantiate a helmchart the user cannot instantiate themselves?)

it will use same permission used today to install the manifests for each samples. Since we are limiting the list of helm charts to install to only s2i charts then user won't be able to install any other charts through samples. Also the sample charts will be coming from Red Hat and will be able to verify digest and signature if one provided.

> displaying a message where?

The message will be displayed on CLI.

Reply:
> it will use same permission used today to install the manifests for each samples. Since we are limiting the list of helm charts to install to only s2i charts then user won't be able to install any other charts through samples. Also the sample charts will be coming from Red Hat and will be able to verify digest and signature if one provided.

ok so it's a specific allow list of charts that can be installed via this api.

> The message will be displayed in logs of the controller

that's going to be pretty difficult for an admin to understand/debug.  Status information on the resource would be a better UX, listing the charts that could not be installed and why not (you can also use this for reporting other failures that might occur while installing a chart)

Reply:
> that's going to be pretty difficult for an admin to understand/debug. Status information on the resource would be a better UX, listing the charts that could not be installed and why not (you can also use this for reporting other failures that might occur while installing a chart)
Thanks for raising the thoughtful questions, we discussed this and are thinking that rather than having the updatable helmchart list to install, how about we have a hardcoded list to install the s2i helmcharts same as it is for current samples, but give the customers the freedom to skip it[skiped helmchart list] as they can do it for current samples. this will avoid all the extra validation/confusion . the customer can skip the helmchart and install a newer version/chart or however they prefer by telling the CSO to not manage it

Reply:
that sounds like a better/less error prone approach to me